meteor/client/ui/Settings/ConfigManifestSettings.tsx

```diff
@@ -80,6 +81,65 @@ function filterLayerMappings(
 	return result
 }
 
+function getFilteredGfxDefaultTableValues<DBInterface extends { _id: ProtectedString<any> }>(
+	item: ConfigManifestEntryFilterDefaultsFromShowMapping<boolean>,
+	configPath: string,
+	object: DBInterface,
+	alternateObject?: any
+): string[] {
+	const sourceAttribute = `${configPath}.${item.sourceTableId}`
+	const targetAttribute = `${configPath}.${item.targetTableId}`
+	const sourceTable = objectPathGet(object, sourceAttribute) ?? objectPathGet(alternateObject, sourceAttribute)
+	const targetTable = objectPathGet(object, targetAttribute) ?? objectPathGet(alternateObject, targetAttribute)
+	const defaultRow = targetTable.find((row) => typeof row === 'object' && row[item.targetCompareColumnId] !== undefined)
+	const result: string[] = []
+	if (!Array.isArray(sourceTable) || !Array.isArray(targetTable) || !defaultRow) {
+		return result
+	}
+
+	sourceTable.forEach((row) => {
+		if (!(typeof row === 'object' && row[item.sourceCollectColumnId] !== undefined)) {
+			return
+		}
+		if (gfxSchemaIsUnavailable(row, item)) {
+			item.sourceCompareColumnId = 'GfxSetup'
+			item.targetCompareColumnId = 'DefaultSetupName'
+		}
+		filterAndCollectGfxDefaultValues(row, item, defaultRow[item.targetCompareColumnId], result)
+	})
+	result.push('N/A')
+	return result
+}
+
+function filterAndCollectGfxDefaultValues(
+	tableRow: any,
+	item: ConfigManifestEntryFilterDefaultsFromShowMapping<boolean>,
+	targetColumnId: string,
+	result: string[]
+) {
+	tableRow[item.sourceCompareColumnId].forEach((compareColumn) => {
+		if (compareColumn === targetColumnId && Array.isArray(tableRow[item.sourceCollectColumnId])) {
+			pushColumnToArray(tableRow, item.sourceCollectColumnId, result)
+		} else if (compareColumn === targetColumnId && !Array.isArray(tableRow[item.sourceCollectColumnId])) {
+			if (!result.includes(tableRow[item.sourceCollectColumnId])) {
+				result.push(tableRow[item.sourceCollectColumnId])
+			}
+		}
+	})
+}
+
+function pushColumnToArray(tableRow: any, columnId: string, resultArray: string[]) {
+	tableRow[columnId].forEach((column) => {
+		if (!resultArray.includes(column)) {
+			resultArray.push(column)
+		}
+	})
+}
+
+function gfxSchemaIsUnavailable(row: any, item: ConfigManifestEntryFilterDefaultsFromShowMapping<boolean>): boolean {
```

Comment:
The type for `row` should be specified.

Reply:
I must say i'm confused as to how i do this, the row comes from the table which we find with `objectPathGet` and that returns `any`.